providers/flagsmith/pkg/provider.go

```diff
@@ -0,0 +1,335 @@
+package flagsmith
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	flagsmithClient "github.com/Flagsmith/flagsmith-go-client/v2"
+	of "github.com/open-feature/go-sdk/pkg/openfeature"
+	"strconv"
+)
+
+type Provider struct {
+	client                  *flagsmithClient.Client
+	usingBooleanConfigValue bool
+}
+
+type ProviderOption func(*Provider)
+
+func NewProvider(client *flagsmithClient.Client, opts ...ProviderOption) *Provider {
+	provider := &Provider{
+		client: client,
+	}
+	for _, opt := range opts {
+		opt(provider)
+	}
+	return provider
+
+}
+
+// flagsmith provider does not have any hooks, returns empty slice
+func (p *Provider) Hooks() []of.Hook {
+	return []of.Hook{}
+}
+
+// Metadata returns value of Metadata (name of current service, exposed to openfeature sdk)
+func (provider *Provider) Metadata() of.Metadata {
```

Comment:
Let's use same receiver name 

```suggestion
func (p *Provider) Metadata() of.Metadata {
```